CI fails to parse response tool calls for gptoss/llama with transformers v5.0.0

CI fails to parse response tool calls with transformers v5:
```python
  FAILED tests/test_chat_template_utils.py::TestParseResponse::test_parse_response_tool_call[gptoss] - assert {'content': '...: 'assistant'} == {'content': '... 'function'}]}
    
    Omitting 1 identical items, use -vv to show
    Differing items:
    {'content': ' to=functions.multiplycommentary json{"a": 3, "b": 4}'} != {'content': ''}
    Right contains 1 more item:
    {'tool_calls': [{'function': {'arguments': {'a': 3, 'b': 4},
                                  'name': 'multiply'},
                     'type': 'function'}]}
    
    Full diff:
      {
    -     'content': '',
    +     'content': ' to=functions.multiplycommentary json{"a": 3, "b": 4}',
          'role': 'assistant',
    -     'tool_calls': [
    -         {
    -             'function': {
    -                 'arguments': {
    -                     'a': 3,
    -                     'b': 4,
    -                 },
    -                 'name': 'multiply',
    -             },
    -             'type': 'function',
    -         },
    -     ],
      }
  FAILED tests/test_chat_template_utils.py::TestParseResponse::test_parse_response_tool_call[llama3.1] - AssertionError: assert {'content': '... 'function'}]} == {'content': '... 'function'}]}
    
    Omitting 2 identical items, use -vv to show
    Differing items:
    {'tool_calls': [{'function': {'arguments': {}, 'name': 'multiply'}, 'type': 'function'}]} != {'tool_calls': [{'function': {'arguments': {'a': 3, 'b': 4}, 'name': 'multiply'}, 'type': 'function'}]}
    
    Full diff:
      {
          'content': '',
          'role': 'assistant',
          'tool_calls': [
              {
                  'function': {
    -                 'arguments': {
    +                 'arguments': {},
    ?                               ++
    -                     'a': 3,
    -                     'b': 4,
    -                 },
                      'name': 'multiply',
                  },
                  'type': 'function',
              },
          ],
      }
  FAILED tests/test_chat_template_utils.py::TestParseResponse::test_parse_response_tool_call[llama3.2] - AssertionError: assert {'content': '... 'function'}]} == {'content': '... 'function'}]}
    
    Omitting 2 identical items, use -vv to show
    Differing items:
    {'tool_calls': [{'function': {'arguments': {}, 'name': 'multiply'}, 'type': 'function'}]} != {'tool_calls': [{'function': {'arguments': {'a': 3, 'b': 4}, 'name': 'multiply'}, 'type': 'function'}]}
    
    Full diff:
      {
          'content': '',
          'role': 'assistant',
          'tool_calls': [
              {
                  'function': {
    -                 'arguments': {
    +                 'arguments': {},
    ?                               ++
    -                     'a': 3,
    -                     'b': 4,
    -                 },
                      'name': 'multiply',
                  },
                  'type': 'function',
              },
          ],
      }
  FAILED tests/test_chat_template_utils.py::TestParseResponse::test_parse_response_tool_call_with_content[gptoss] - assert {'content': '...: 'assistant'} == {'content': "... 'function'}]}
    
    Omitting 1 identical items, use -vv to show
    Differing items:
    {'content': 'analysisLet\'s call the tool.assistant to=functions.multiplycommentary json{"a": 3, "b": 4}'} != {'content': "Let's call the tool."}
    Right contains 1 more item:
    {'tool_calls': [{'function': {'arguments': {'a': 3, 'b': 4},
                                  'name': 'multiply'},
                     'type': 'function'}]}
    
    Full diff:
      {
    -     'content': "Let's call the tool.",
    +     'content': "analysisLet's call the tool.assistant to=functions.multiplycommentary "
    +     'json{"a": 3, "b": 4}',
          'role': 'assistant',
    -     'tool_calls': [
    -         {
    -             'function': {
    -                 'arguments': {
    -                     'a': 3,
    -                     'b': 4,
    -                 },
    -                 'name': 'multiply',
    -             },
    -             'type': 'function',
    -         },
    -     ],
      }
  FAILED tests/test_chat_template_utils.py::TestParseResponse::test_parse_response_tool_call_without_arguments[gptoss] - AssertionError: assert {'content': '...: 'assistant'} == {'content': '... 'function'}]}
    
    Omitting 1 identical items, use -vv to show
    Differing items:
    {'content': ' to=functions.pingcommentary json{}'} != {'content': ''}
    Right contains 1 more item:
    {'tool_calls': [{'function': {'arguments': {}, 'name': 'ping'},
                     'type': 'function'}]}
    
    Full diff:
      {
    -     'content': '',
    +     'content': ' to=functions.pingcommentary json{}',
          'role': 'assistant',
    -     'tool_calls': [
    -         {
    -             'function': {
    -                 'arguments': {},
    -                 'name': 'ping',
    -             },
    -             'type': 'function',
    -         },
    -     ],
      }
  FAILED tests/test_chat_template_utils.py::TestParseResponse::test_parse_response[gptoss] - AttributeError: 'NoneType' object has no attribute 'get'
  FAILED tests/test_chat_template_utils.py::TestAddResponseSchema::test_add_response_schema[gptoss] - ValueError: Node has JSON parser but could not parse its contents as JSON. You can use the `allow_non_json` parser arg for nodes which may contain JSON or string content.
  
  Content: to=functions.multiply<|channel|>commentary json<|message|>{"a": 3, "b": 4}<|call|>
  
  Error: Expecting value: line 1 column 1 (char 0)
```

--- a/tests/test_chat_template_utils.py
+++ b/tests/test_chat_template_utils.py
@@ -124,7 +124,15 @@ class TestAddResponseSchema:
         "tokenizer_name",
         [
             pytest.param("trl-internal-testing/tiny-Glm4MoeForCausalLM", id="glm4moe"),
-            pytest.param("trl-internal-testing/tiny-GptOssForCausalLM", id="gptoss"),
+            pytest.param(
+                "trl-internal-testing/tiny-GptOssForCausalLM",
+                id="gptoss",
+                marks=pytest.mark.xfail(
+                    Version(transformers.__version__) < Version("5.5.0"),
+                    reason="Upstream bug in response parsing (see #5753; fixed in transformers#45166)",
+                    strict=True,
+                ),
+            ),
             pytest.param("trl-internal-testing/tiny-LlamaForCausalLM-3.1", id="llama3.1"),
             pytest.param("trl-internal-testing/tiny-LlamaForCausalLM-3.2", id="llama3.2"),
             pytest.param("trl-internal-testing/tiny-Qwen2ForCausalLM-2.5", id="qwen2.5"),
@@ -709,6 +717,10 @@ class TestParseResponse:
         return processing_class
 
     def test_parse_response(self, model_name):
+        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM") and Version(
+            transformers.__version__
+        ) < Version("5.5.0"):
+            pytest.skip("Upstream bug in response parsing (see #5753; fixed in transformers#45166)")
         processing_class = self._load(model_name)
         messages = [
             {"role": "user", "content": "What is 3*4?"},
@@ -762,6 +774,12 @@ class TestParseResponse:
         assert parsed == expected
 
     def test_parse_response_tool_call(self, model_name):
+        if model_name in (
+            "trl-internal-testing/tiny-GptOssForCausalLM",
+            "trl-internal-testing/tiny-LlamaForCausalLM-3.1",
+            "trl-internal-testing/tiny-LlamaForCausalLM-3.2",
+        ) and Version(transformers.__version__) < Version("5.5.0"):
+            pytest.skip("Upstream bug in response parsing (see #5753; fixed in transformers#45166)")
         processing_class = self._load(model_name)
         tool_calls = [{"type": "function", "function": {"name": "multiply", "arguments": {"a": 3, "b": 4}}}]
         messages = [
@@ -794,6 +812,10 @@ class TestParseResponse:
             "trl-internal-testing/tiny-LlamaForCausalLM-3.2",
         ):
             pytest.skip("Llama 3.1 / 3.2 templates only allow a single tool call per assistant turn, with no content.")
+        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM") and Version(
+            transformers.__version__
+        ) < Version("5.5.0"):
+            pytest.skip("Upstream bug in response parsing (see #5753; fixed in transformers#45166)")
         processing_class = self._load(model_name)
         tool_calls = [{"type": "function", "function": {"name": "multiply", "arguments": {"a": 3, "b": 4}}}]
         messages = [
@@ -815,6 +837,10 @@ class TestParseResponse:
         assert parsed == expected
 
     def test_parse_response_tool_call_without_arguments(self, model_name):
+        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM") and Version(
+            transformers.__version__
+        ) < Version("5.5.0"):
+            pytest.skip("Upstream bug in response parsing (see #5753; fixed in transformers#45166)")
         processing_class = self._load(model_name)
         tool_calls = [{"type": "function", "function": {"name": "ping", "arguments": {}}}]
         messages = [

--- a/tests/test_chat_template_utils.py
+++ b/tests/test_chat_template_utils.py
@@ -717,7 +717,7 @@ class TestParseResponse:
         return processing_class
 
     def test_parse_response(self, model_name):
-        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM") and Version(
+        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM",) and Version(
             transformers.__version__
         ) < Version("5.5.0"):
             pytest.skip("Upstream bug in response parsing (see #5753; fixed in transformers#45166)")
@@ -812,7 +812,7 @@ class TestParseResponse:
             "trl-internal-testing/tiny-LlamaForCausalLM-3.2",
         ):
             pytest.skip("Llama 3.1 / 3.2 templates only allow a single tool call per assistant turn, with no content.")
-        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM") and Version(
+        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM",) and Version(
             transformers.__version__
         ) < Version("5.5.0"):
             pytest.skip("Upstream bug in response parsing (see #5753; fixed in transformers#45166)")
@@ -837,7 +837,7 @@ class TestParseResponse:
         assert parsed == expected
 
     def test_parse_response_tool_call_without_arguments(self, model_name):
-        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM") and Version(
+        if model_name in ("trl-internal-testing/tiny-GptOssForCausalLM",) and Version(
             transformers.__version__
         ) < Version("5.5.0"):
             pytest.skip("Upstream bug in response parsing (see #5753; fixed in transformers#45166)")

